src/dxbc/dxbc_compiler.cpp

```diff
@@ -4745,6 +4747,13 @@ namespace dxvk {
       getPointerTypeId(info),
       m_xRegs.at(regId).varId,
       1, &vectorId.id);
+
+    if (boundsCheck) {
+      result.boundsCheck = m_module.opULessThan(
+        m_module.defBoolType(), vectorId.id,
+        m_module.constu32(m_xRegs.at(regId).alength));
+    }
```

Comment:
Probably better to move this into emitRegisterStore instead so that we don't need the new struct member.

Reply:
How should I do that?

Should I add a `emitIndexLoad(operand.idx[1])` in `emitRegisterStore`, create a `getIndexableTempPtr(const DxbcRegister& operand, DxbcRegisterValue vectorId)` helper or add a output `vectorId` parameter to `emitGetIndexableTempPtr`?

Reply:
Actually, don't we have to consume the OpAccessChain in the same block that contains the instruction for the SPIR-V to be valid?

Reply:
I don't see any requirement in the SPIR-V spec. There is a requirement that OpSampledImage instructions are in the same block which they are consumed though

Reply:
Right, that shouldn't be much of a concern then (at least in theory, in practice I wouldn't want to rely on it since drivers tend to be buggy with code patterns that aren't generated by glslang)

> Should I add a emitIndexLoad(operand.idx[1]) in emitRegisterStore, create a getIndexableTempPtr(const DxbcRegister& operand, DxbcRegisterValue vectorId) helper

I suppose that's the better option.